mdal/frmts/mdal_hdf5.cpp

```diff
@@ -7,18 +7,31 @@
 #include <cstring>
 
 
-HdfFile::HdfFile( const std::string &path )
-  : d( std::make_shared< Handle >( H5Fopen( path.c_str(), H5F_ACC_RDONLY, H5P_DEFAULT ) ) )
+HdfFile::HdfFile( const std::string &path, bool writeNew )
```

Comment:
I do not like the new argument writeNew in the classes. I would prefer to either create the createFile/Attribute/... function outside the classes and then use the classes the old way?

In general in C++, you do not want constructors that could potentially fail/raise error.

Reply:
I agree that this is far from ideal, I didnt want to break existing code and this kept other classes using Hdf classes working without a change. How to refactor this is probably for deeper discussion.